src/modelmanager.cpp

```diff
@@ -825,12 +825,22 @@ void ModelManager::watcher(std::future<void> exitSignal) {
             loadConfig(configFilename);
         }
         updateConfigurationWithoutConfigFile();
+        cleanupResources();
```

Comment:
Currently resources will never be freed if watcher interval is 0. We can either clean up resource immediately if interval is 0 or create new thread for this purpose which interval should not be modifiable.

Reply:
The proposed solution is to implement cleanerThread that would be responsible for cleaning up resources, but also sequences